Project/Assets/ML-Agents/Examples/Crawler/Scripts/CrawlerAgent.cs

```diff
@@ -184,64 +281,93 @@ void FixedUpdate()
                 : unGroundedMaterial;
         }
 
+        var cubeForward = m_OrientationCube.transform.forward;
+
         // Set reward for this step according to mixture of the following elements.
-        if (rewardMovingTowardsTarget)
-        {
-            RewardFunctionMovingTowards();
-        }
+        // a. Match target speed
+        //This reward will approach 1 if it matches perfectly and approach zero as it deviates
+        var matchSpeedReward = GetMatchingVelocityReward(cubeForward * TargetWalkingSpeed, GetAvgVelocity());
 
-        if (rewardFacingTarget)
+        //Check for NaNs
+        if (float.IsNaN(matchSpeedReward))
         {
-            RewardFunctionFacingTarget();
+            throw new ArgumentException(
```

Comment:
I missed that on my first pass, but I do not think this is really an ArgumentException.

Reply:
I agree. @chriselion do you think we can get rid of these NaN checks? have you seen any more recently? If not I'll just take them out.

Reply:
Yeah, think they're OK to remove now.

Reply:
removed